### DISCUSSION TITLE
Cozo for Enterprise?

### DISCUSSION THREAD

**Discussion by barbinbrad** [cozodb/cozo#10] [Q&A] [answered]
Hey! Really impressive work! I do a lot of work on the frontend, so please forgive me if these questions seem stupid. 

A very common problem in the enterprise software space is the need for groups that can have members or other groups, and then resolving those groups on the fly to figure out whether a person has access to a resource, or whether a person should receive a message, etc. Google has a service called Zanzibar to solve this problem. But for most of us in the enterprise space, either don't know how to solve this problem, or don't have the resources to solve it correctly.

So I guess my questions would be:

- Would cozo be a good tool to solve the problem I've described?
- If so, does it lend itself to large-scale deployment practices like replications and backups, or is it more of SQLite-2019-type of thing?

**Comment by zh217** [comment, marked as answer]
Hi, thanks for your interest!

Regarding your first question, yes, Cozo has what you described as one of its main use cases (https://github.com/cozodb/cozo#use-cases). You store your data and metadata about people, roles, resources etc. as stored relations, and the rules for whether a person has access are encoded as CozoScript queries. If resources permit, we may write a tutorial-style document showing how to do it step-by-step!

For your second question, Cozo is probably too young for enterprise deployment, and I would suggest you wait until 1.0 release for using it for any work dealing with critical data. Backups and replications are currently not implemented but will be. Cozo is only half a year old, and I don't think the enterprise people will allow you to use anything this young even if we claim it to be "production-ready" (it certainly is not)!
